### PR TITLE
Improved error message for invalid arguments [ECR-2841]

### PR DESCRIPTION
**Pull request by skletsun**
## Overview

Added human-readable error descriptions for JNI codes that weren't covered in the jni.rs lib for the result of JVM creation.

---
See: https://jira.bf.local/browse/ECR-2841

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [x] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has Javadoc
- [x] Method preconditions are checked and documented in the Javadoc of the method
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [x] The continuous integration build passes

### CODE REVIEW

File: exonum-java-binding/core/rust/src/runtime/java_service_runtime.rs [hunk 1]
```diff
@@ -237,3 +257,59 @@ impl JavaServiceRuntime {
         }))
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn transform_jni_error_error_type_other() {
```
Comment (vitvakatu) [inline exonum-java-binding/core/rust/src/runtime/java_service_runtime.rs]: Don't you trust LLVM in making pattern matching? :)

Comment (skletsun) [reply exonum-java-binding/core/rust/src/runtime/java_service_runtime.rs]: I do, but I don't trust myself in writing `match` blocks 😄

File: exonum-java-binding/core/rust/src/runtime/java_service_runtime.rs [hunk 2]
```diff
@@ -131,7 +132,26 @@ impl JavaServiceRuntime {
...
+        jni::JavaVM::new(args)
+            .map_err(Self::transform_jni_error)
+            .expect("Unable to create JVM")
+    }
+
+    /// Transforms JNI errors by converting JNI error codes of error of type `Other` to its string
+    /// representation.
+    fn transform_jni_error(error: Error) -> Error {
+        match error.0 {
+            ErrorKind::Other(code) => match code {
```
Comment (dmitry-timofeev) [inline exonum-java-binding/core/rust/src/runtime/java_service_runtime.rs]: I wonder if it makes sense to upstream anything :thinking: ?

File: exonum-java-binding/core/rust/src/runtime/java_service_runtime.rs (hunk 1 above)
Comment (dmitry-timofeev) [reply exonum-java-binding/core/rust/src/runtime/java_service_runtime.rs]: I agree that it seems to be too easy to break, so we can reduce the number of tests to three (a single known error code, unknown error code, other type of error).

File: exonum-java-binding/core/rust/src/runtime/java_service_runtime.rs (hunk 2 above)
Comment (skletsun) [reply exonum-java-binding/core/rust/src/runtime/java_service_runtime.rs]: Potential improvement - enhance the `jni::errors::jni_error_code_to_result()` function by making it translate every JNI code to separate error (current behavior is kinda minimal implementation that translates `JNI_OK` and `thread detached` to Result::Ok and ErrorKind::ThreadDetached respectively, any other JNI error code -> ErrorKind::Other(error_code)). Pros: error messages come for free; Cons: it's obviously a breaking change for those who already rely on ErrorKind::Other(error_code).

Comment (dmitry-timofeev) [reply exonum-java-binding/core/rust/src/runtime/java_service_runtime.rs]: On a quick look, it seems to make sense! I'd submit an issue to jni-rs or our issue tracker.

Comment (skletsun) [reply exonum-java-binding/core/rust/src/runtime/java_service_runtime.rs]: https://github.com/jni-rs/jni-rs/issues/166

**Pull request merged**